CODING_GUIDELINES.md

```diff
@@ -6,30 +6,86 @@ This document contains the coding guidelines for the ASPIRE Python project.
 
+In the future compoenents of PEP8 and PEP257 will be checked programmatically.  Documentation of a standard tool to perform
+these operations locally will be provided to developers at that time.
 
-## Best Practices
+## Best Practices Reading
 
 For beginners in scientific computing, some general guides are suggested in Greg Wilson's papers as below:
 
- 1. [Good Enough Practices in Scientific Computing]( https://doi.org/10.1371/journal.pcbi.1005510) 
- 2. [Best Practices for Scientific Computing]( https://doi.org/10.1371/journal.pbio.1001745) 
-
+ 1. [Good Enough Practices in Scientific Computing]( https://doi.org/10.1371/journal.pcbi.1005510)
+ 2. [Best Practices for Scientific Computing]( https://doi.org/10.1371/journal.pbio.1001745)
 
 ## Source Code Control
 
-As an open source software, we use Git and GitHub for our source code control. 
-The following links are useful tutorials. 
+As an open source software, we use Git with GitHub for our source code control.
+The following links are useful tutorials.
 1. Git tutorials from [Atalassian](https://www.atlassian.com/git/tutorials) and [Git-SCM](https://git-scm.com/docs/gittutorial)
-2. GitHub guides from [GitHub](https://guides.github.com/) 
+2. GitHub guides from [GitHub](https://guides.github.com/)
 
-## Git Workflow Changes
+## Git Workflow
 
-Our project follows [Gitflow Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).
+Our project follows a reduced [Gitflow Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).
 Please submit any PRs against the `develop` branch.
 
 ![Gitflow Diagram](https://wac-cdn.atlassian.com/dam/jcr:61ccc620-5249-4338-be66-94d563f2843c/05%20(2).svg?cdnVersion=357)
 
+The basic idea is that we have `feature` branches,  `develop` as an integration focused branch, and `master` for releases.
+We do not currently have a need for a dedicated release or staging branch; in our case `develop` also serves this purpose.
+Generally feature branches should branch off of the latest `develop` branch.
+
+Developers are welcome to maintain their own fork, but this is not required.
+
+When `develop` is at a milestone where we want to cut a release we will commit with a PEP440 compliant `Major.Minor` version
+scheme, tag it similarly, and complete a merge to master following the controls below. It is at this point the
+release would be a candadite to upload to PyPI.
+
+###  Git Workflow Controls
+
+To facilitate effective integration, merges into `develop` must occur as a pull request that is approved by two reviewers.
+One reviewer can be anyone, but should probably be someone related to the effort if that is applicable.
+The second developer is a designated codeowner described in the `CODEOWNERS` file.
+
+Additionally, all outstanding requested changes and comments must be resolved before merging.  Pushing new changes should trigger
+a re-review process of changed files to prevent oversights.
```

Comment:
Would it make sense to require comments to be resolved by the reviewer? I frequently find myself unhiding conversations that have been resolved to see what the resolution was. Normally, it's fine, but I think this would make it clearer when an issue is resolved. Maybe this would be too cumbersome.

Reply:
I don't think that is something that can be enforced with github controls. If things are that bad, we're in trouble.

Reply:
If it is possible to do so, that would make a lot of sense...

Reply:
Not something that would need to be enforced by GH, just something that we agree to adhere to.

Reply:
I'm good with either way.

Usually if I can get the automata to enforce it, there is less to think about; that's all I was going for there.

Reply:
2c25d22